proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/resources/HubResponseTranslatorResource.java

```diff
@@ -88,11 +86,12 @@ public Response failureResponse(SamlFailureResponseGenerationRequest failureResp
         return Response.ok().entity(samlMessage).build();
     }
 
+    // TODO: Remove once we set all the headers correctly
     private void logSamlResponse(org.opensaml.saml.saml2.core.Response samlResponse) {
-            proxyNodeLogger.addContext(ProxyNodeMDCKey.EIDAS_REQUEST_ID, Objects.requireNonNullElse(samlResponse.getInResponseTo(), ""));
-            proxyNodeLogger.addContext(ProxyNodeMDCKey.EIDAS_DESTINATION, Objects.requireNonNullElse(samlResponse.getDestination(), ""));
-            proxyNodeLogger.addContext(ProxyNodeMDCKey.EIDAS_ISSUER, samlResponse.getIssuer() != null ? samlResponse.getIssuer().getValue() : "");
-            proxyNodeLogger.log(Level.INFO, EIDAS_RESPONSE_LOGGER_MESSAGE);
-
-        }
+        LOG.addContext(ProxyNodeMDCKey.EIDAS_REQUEST_ID, Objects.requireNonNullElse(samlResponse.getInResponseTo(), ""));
```

Comment:
How do these values in the MDC get cleaned up after logging?

Reply:
In the ProxyNodeLogging filter:

for (ProxyNodeMDCKey key : ProxyNodeMDCKey.values()) {
            MDC.remove(key.name());
        }
We clean everything up before sending the response back